encoding/protobuf.go

```diff
@@ -0,0 +1,96 @@
+package encoding
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/jhump/protoreflect/desc"
+	"github.com/jhump/protoreflect/dynamic"
+	"github.com/yarpc/yab/protobuf"
+	"github.com/yarpc/yab/transport"
+)
+
+type protoSerializer struct {
+	serviceName string
+	methodName  string
+	method      *desc.MethodDescriptor
+}
+
+// NewProtobuf returns a protobuf serializer.
+func NewProtobuf(fullMethodName string, source protobuf.DescriptorProvider) (Serializer, error) {
+	svc, mth, err := splitMethod(fullMethodName)
+	if err != nil {
+		return nil, err
+	}
+	dsc, err := source.FindSymbol(svc)
+	if err != nil {
+		return nil, fmt.Errorf("failed to query for service for symbol %q: %v", svc, err)
+	}
+	sd, ok := dsc.(*desc.ServiceDescriptor)
+	if !ok {
+		return nil, fmt.Errorf("target server does not expose service %q", svc)
+	}
+	mtd := sd.FindMethodByName(mth)
```

Comment:
Distinguishing `mtd` / `mth` and `dsc` / `desc` don’t seem to follow any particular rule. Perhaps we should abandon brevity. If I were to guess:

- `svc`: `serviceName`
- `mth`: `methodName`
- `dsc`: `serviceSymbol`
- `sd`: `serviceDescriptor`
- `mtd`: `methodDescriptor`